lib/logstash/config/defaults.rb

```diff
@@ -0,0 +1,9 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "concurrent"
+
+module LogStash module Config module Defaults
+  def self.cpu_cores
+    Concurrent.processor_count
```

Comment:
@jordansissel - Concurrent does provide a physical_processor_count method but on the JVM both methods return the hyperthreads.